src/Build/Evaluation/LazyItemEvaluator.cs

```diff
@@ -359,9 +404,26 @@ private static ImmutableList<ItemData>.Builder ComputeItems(LazyItemList lazyIte
                     currentList._memoizedOperation.Apply(items, currentGlobsToIgnore);
                 }
 
+                ProcessNonWildCardItemUpdates(itemsWithNoWildcards, items);
+
                 return items;
             }
 
+            private static void ProcessNonWildCardItemUpdates(Dictionary<string, UpdateOperation> itemsWithNoWildcards, ImmutableList<ItemData>.Builder items)
```

Comment:
Thinking out loud: Should this function check if `itemsWithNoWildcards` somehow mistakenly receives an item with a wildcard? It should be fine considering it's a private function.

Reply:
It could...I don't have a strong opinion. If there is a fragment with a wild card, I'd suggest failing hard, but it's a private method that I don't anticipate being used elsewhere with a very clear name (NoWildcards) so we shouldn't have a problem. It would also be a perf hit, since we'd be checking every fragment. On the other hand, safe is generally better than sorry, and someone could theoretically mangle this in the future.